src/components/profile/Profile.js

```diff
@@ -8,24 +8,29 @@ import RecoveryContainer from './Recovery/RecoveryContainer'
 import HardwareDevices from './hardware_devices/HardwareDevices'
 import TwoFactorAuth from './two_factor/TwoFactorAuth'
 import { LOADING, NOT_FOUND, useAccount } from '../../hooks/allAccounts'
-import { get2faMethod, getLedgerKey, checkCanEnableTwoFactor, getAccessKeys } from '../../actions/account';
+import { getLedgerKey, checkCanEnableTwoFactor, getAccessKeys } from '../../actions/account';
 
 export function Profile({ match }) {
     const loginAccountId = useSelector(state => state.account.accountId)
+    const recoveryMethods = useSelector(({ recoveryMethods }) => recoveryMethods);
     const accountId = match.params.accountId || loginAccountId
     const isOwner = accountId === loginAccountId
     const account = useAccount(accountId)
     const dispatch = useDispatch();
+    const twoFactor = recoveryMethods[account.accountId] && recoveryMethods[account.accountId].filter(m => m.kind.includes('2fa'))[0]
 
     useEffect(() => { 
         if (isOwner) {
-            dispatch(getAccessKeys(accountId))
-            dispatch(getLedgerKey())
-            dispatch(get2faMethod())
-            dispatch(checkCanEnableTwoFactor(account))
+            getInitialData()
         }
     }, []);
 
+    const getInitialData = async () => {
+        await dispatch(getAccessKeys(accountId))
```

Comment:
@Patrick1904 is it intended that we are waiting for each of these to finish before making another request?